File: app/src/main/kotlin/com/spacex/ui/InteractionsHistory.java
package com.spacex.ui;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class InteractionsHistory {

  private static final String wasWelcomeShown = "was-welcome-shown";

  private static final SharedPreferences sSharedPreferencesDefault =
      PreferenceManager.getDefaultSharedPreferences(RocketLaunchesApp.getInstance());

  static boolean wasWelcomeShown() {
    return sSharedPreferencesDefault.getBoolean(wasWelcomeShown, false);
  }

  public static void setWelcomeAsShown() {
    sSharedPreferencesDefault.edit().putBoolean(wasWelcomeShown, true).apply();
  }
}

File: app/src/main/kotlin/com/spacex/ui/InteractionsHistory.kt
package com.spacex.ui

import android.preference.PreferenceManager

internal object InteractionsHistory {

    private const val wasWelcomeShown = "was-welcome-shown"
    private val sSharedPreferencesDefault = PreferenceManager.getDefaultSharedPreferences(RocketLaunchesApp.instance)

    fun wasWelcomeShown(): Boolean {
        return sSharedPreferencesDefault.getBoolean(wasWelcomeShown, false)
    }

    fun setWelcomeAsShown() {
        sSharedPreferencesDefault.edit().putBoolean(wasWelcomeShown, true).apply()
    }
}
